crates/c-api/src/wasi.rs

```diff
@@ -270,3 +327,89 @@ pub unsafe extern "C" fn wasi_config_preopen_dir(
 
     true
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn wasi_pipe_new(
+    limit: usize,
+    ret_read_pipe: *mut *mut wasi_read_pipe_t,
+    ret_write_pipe: *mut *mut wasi_write_pipe_t,
```

Comment:
I think this can be `&mut *mut wasi_write_pipe_t` which should remove the need for this function to be unsafe.

Reply:
Is it possible to "null check" a `&mut` since it's expected to be a valid reference ?

If not, want would append if the function is given a null pointer ? UB?

Or, is a function documentation enough to be confident than the function is used properly ?

Reply:
My point is that the `&mut` layer is part of the contract of the function, it must be valid. What it points to could be invalid, hence the next layer of raw pointers.